draft-ietf-quic-tls.md

```diff
@@ -866,6 +867,85 @@ the connection that is hosted on stream 0.  This sequence number is not visible
 to QUIC.
 
 
+## Packet Number Protection {#pn-encrypt}
```

Comment:
If we move to varint packet numbers inside encryption, determining the length of the varint with this approach seems not completely obvious.  If we stick with this approach instead of the overlapping crypto operations, it would be good to add small section on removing packet number protection.

Reply:
> Q: If we encode the packet number length inside the encryption, which seems nice, does the decrypter have to assume the longest possible packet number length, then check the first byte after decryption to determine the actual length?

Since the encryption is XOR based, it suffice to decrypt the first byte with XOR to decide the length, or, just the first two bits, actually. In praxis it might be simpler/faster to XOR with the maximum length, then read as necessary. Escpecially if the packet is guaranteed to be long enought and when fast unaligned 8 byte reads are available.

Reply:
What @mikkelfj says.  Say we choose an encoding with a 4 octet maximum length, then you copy 4 octets out, XOR those, then decode from that buffer.  You then know the length and can discard the excess.

Reply:
How about something like:
1) Set a barrier at the start of the chosen ciphertext input.
2) If the barrier is further than 8 bytes from the beginning of the packet number, XOR 8 bytes from beginning of packet number, 
3) If the barrier is closer than that, XOR up to the barrier.

Reply:
So this isn't missed by people reviewing, I've taken this suggestion.  That way we can make stronger assertions about birthday bounds.